### ISSUE TITLE
[Bug Report]: ReferenceError: require is not defined when use builtins.provide 

### ISSUE THREAD

**Issue by 9aoy** [web-infra-dev/rspack#2590]
### System Info

  System:
    OS: macOS 12.5.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 116.39 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.15.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.5.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 111.0.5563.146
    Safari: 15.6.1
  npmPackages:
    @rspack/cli: 0.1.5 => 0.1.5 

### Details

I want use builtins.provide to auto load `process`, but I get a runtime error.

```
	builtins: {
		provide: {
                     process: [nodeLibsBrowser.process],
		}
	},
```

<img width="529" alt="image" src="https://user-images.githubusercontent.com/22373761/229451361-f3e8bbed-721c-4540-aef7-35459132906d.png">

It seems that the module should not be loaded when identifier already define.

<img width="1201" alt="image" src="https://user-images.githubusercontent.com/22373761/229451478-9234c165-3e0a-43a0-bfef-df5406ed76ee.png">

![image](https://user-images.githubusercontent.com/22373761/229452029-ddc4852c-feab-41da-a17f-a31bc4e92f7a.png)

### Reproduce link

https://github.com/9aoy/rspack-issue-reproduce/blob/main/builtins-provide/rspack.config.js

### Reproduce Steps

1. npm install
2. npm run dev
3. open the browser

**Comment by suxin2017** [comment]
Can you assign this problem to me? I will have time to deal with it this weekend

**Comment by hyf0** [comment]
Fixed by #2721.
